2D Simulation/game.js

```diff
@@ -335,6 +345,9 @@ function send_user_message_to_server(gameConfig) {
     if (start_button_pressed) {
         if (movesCorrector.handle_message(message)) {
             return;
+        } else if (movesCorrector.is_handling_ambiguity()) {
+            movesCorrector.finish_handling_ambiguity(message);
```

Comment:
I feel like this should go inside the `handle_message` method. Let the `movesCorrector` figure out which state it's in, not the calling code.

Reply:
I disagree. Calling `finish_handling_ambiguity` seems bad if it isn't going to handle any ambiguity. That seems like weird control flow to me.

Reply:
I'm more talking about moving lines 322-323 into the `handle_message` method. Everything else would stay the same; `finish_handling_ambiguity` could then be a private method.

Reply:
I guess I misunderstood the purpose of that method. I can move it in there.